main.lua

```diff
@@ -57,112 +60,128 @@ Ball = { -- single object
     dVelocity = 300,
 }
 
-Brick = { -- generic sprite used by multiple brick objects
+-- template for bricks
+local brickTemplate = {
     imagePath = "assets/png/element_grey_rectangle.png",
     spriteName = "brick_grey",
     width = 64,
     height = 32,
 }
 
-Bricks = {} -- list of actual brick objects used in game
+-- list of actual brick objects used in game
+local bricks = {}
 
-NextCollision = {
-    position = false,
-    object = false,
-}
-
-function CreateBrick(centerX, centerY, rotation)
-    local position = Vector.new(centerX - Brick.width / 2, centerY - Brick.height / 2)
+local function createBrick(x, y, rotation)
+    assert(type(x) == "number")
+    assert(type(y) == "number")
+    if rotation ~= nil then assert(type(rotation) == "number") else rotation = 0.0 end
 
     return {
-        spriteName = "brick_grey",
-        width = Brick.width,
-        height = Brick.height,
-        position = position,
+        spriteName = brickTemplate.spriteName,
+        width = brickTemplate.width,
+        height = brickTemplate.height,
+        position = Vector.new(x, y),
         rotation = rotation,
     }
 end
 
-function ResetGame()
-    -- setup paddle
-    Paddle.position = Vector.new((Window.width - Paddle.width) / 2, Window.height - 100)
-    Paddle.velocity = Vector.new()
-    Paddle.acceleration = Vector.new()
+local function resetBall()
+    ball.position:setXY(window.width / 2, paddle.position.y - (ball.height + 8))
+    ball.velocity:setXY(0, 1)
+end
+
+local function resetPaddle()
+    paddle.position:setXY(window.width / 2, window.height - 100)
+    paddle.velocity:setXY(0, 0)
+    paddle.acceleration:setXY(0, 0)
 
-    -- check for keys pressed before game setup
-    if love.keyboard.isDown(KeyEnum.LEFT) then
-        Paddle.acceleration:transform(LeftVector)
-    end
-    if love.keyboard.isDown(KeyEnum.RIGHT) then
-        Paddle.acceleration:transform(RightVector)
-    end
+    -- check for keys pressed and not released before resetting
+    if love.keyboard.isDown(KeyEnum.LEFT) then paddle.acceleration:transform(left) end
```

Comment:
Or
```lua
love.keyboard.isDown(KeyEnum.LEFT) and paddle.acceleration:transform(left)
```

Less keywords but probably also less readable. What do you think?

Reply:
Looks fine, but I'd rather focus on code readability 😄